src/game/default/g_main.c

```diff
@@ -796,21 +799,23 @@ static void G_Frame(void) {
 			return;
 		}
 	}
+		
+	if (!TIMEOUT) {
```

Comment:
Hah! That's awesome. I don't think reverting `g_level.frame_num` is dangerous. `g_level.frame_num` is not a protocol-level frame number (like `sv_server.frame_num`).